### ISSUE TITLE
Contacts (4.0.7) doesn't show any more contacts under NC 22.2.4

### ISSUE THREAD

**Issue by JimmyKater** [nextcloud/contacts#2638]
### Describe the bug
Since several weeks (sure enough more than 12) I can't access my personal contacts anymore. Neither via the WebUI (it just says "no contacts") nor via external Tools (like Outlook Sychronizer - it throws me a 500 error)
In the meanwhile I'v upgraded NC server to 22.2.4 (from .3) hoping that it would solve the problem. And I can't remember clearly if I had updated my contacts version to v4.0.7 or if it happend even with v4.0.6 - could be that this error took place under 4.0.6 as well.
Important notice: I CAN see my contacts if I'd click the contacts icon in menubar in upper right corner.

**To Reproduce**
right now everytime I goto the contacts-app section under my WebGUI I never see any contacts there. But I dunno what I did to make this happen.

### Server configuration
Operating system: Linux 5.4.0-92-generic #103-Ubuntu SMP Fri Nov 26 16:13:00 UTC 2021 x86_64

Webserver: Apache (fpm-fcgi)

Database: pgsql PostgreSQL 12.9 (Ubuntu 12.9-0ubuntu0.20.04.1) on x86_64-pc-linux-gnu, compiled by gcc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0, 64-bit

PHP version: 7.4.3

Modules loaded: Core, date, libxml, openssl, pcre, zlib, filter, hash, Reflection, SPL, session, standard, sodium, cgi-fcgi, PDO, xml, bcmath, bz2, calendar, ctype, curl, dom, mbstring, FFI, fileinfo, ftp, gd, gettext, gmp, iconv, igbinary, imap, intl, json, ldap, exif, pdo_pgsql, pgsql, Phar, posix, readline, redis, shmop, SimpleXML, smbclient, soap, sockets, sysvmsg, sysvsem, sysvshm, tokenizer, xmlreader, xmlwriter, xsl, zip, libsmbclient, Zend OPcache

Nextcloud version: 22.2.4 - 22.2.4.3

Updated from an older Nextcloud/ownCloud or fresh install: 22.0.3

Where did you install Nextcloud from: it's a VM from HanssonIT

<details><summary>Signing status</summary>

Array ( )

</details>

<details><summary>List of activated apps</summary>

Enabled:
 - accessibility: 1.8.0
 - activity: 2.15.0
 - admin_audit: 1.12.0
 - apporder: 0.14.0
 - audioplayer: 3.2.4
 - bbb: 2.2.0
 - bookmarks: 10.0.3
 - calendar: 3.0.5
 - carnet: 0.24.4
 - cloud_federation_api: 1.5.0
 - cms_pico: 1.0.17
 - comments: 1.12.0
 - contacts: 4.0.7
 - contactsinteraction: 1.3.0
 - cookbook: 0.9.9
 - dashboard: 7.2.0
 - dav: 1.20.0
 - deck: 1.5.5
 - event_update_notification: 1.4.0
 - federatedfilesharing: 1.12.0
 - federation: 1.12.0
 - files: 1.17.0
 - files_downloadactivity: 1.12.0
 - files_external: 1.13.1
 - files_pdfviewer: 2.3.1
 - files_rightclick: 1.1.0
 - files_sharing: 1.14.0
 - files_trackdownloads: 1.11.0
 - files_videoplayer: 1.11.0
 - group_default_quota: 0.1.3
 - groupfolders: 10.0.2
 - groupquota: 0.1.7
 - impersonate: 1.9.0
 - integration_dropbox: 1.0.4
 - logreader: 2.7.0
 - lookup_server_connector: 1.10.0
 - maps: 0.1.10
 - mediadc: 0.1.8
 - metadata: 0.15.0
 - nextcloud_announcements: 1.11.0
 - notifications: 2.10.1
 - notify_push: 0.3.0
 - oauth2: 1.10.0
 - password_policy: 1.12.0
 - passwords: 2021.12.20
 - phonetrack: 0.7.0
 - photos: 1.4.0
 - privacy: 1.6.0
 - provisioning_api: 1.12.0
 - quota_warning: 1.13.0
 - rainloop: 7.1.3
 - richdocuments: 4.2.4
 - richdocumentscode: 21.11.103
 - sendent: 1.2.10
 - serverinfo: 1.12.0
 - settings: 1.4.0
 - sharebymail: 1.12.0
 - side_menu: 2.3.3
 - socialsharing_diaspora: 2.4.0
 - socialsharing_email: 2.4.0
 - socialsharing_facebook: 2.4.0
 - socialsharing_twitter: 2.4.0
 - spreed: 12.2.3
 - support: 1.5.0
 - suspicious_login: 4.1.0
 - systemtags: 1.12.0
 - text: 3.3.0
 - theming: 1.13.0
 - theming_customcss: 1.10.0
 - twofactor_backupcodes: 1.11.0
 - twofactor_webauthn: 0.2.15
 - updatenotification: 1.12.0
 - user_saml: 4.1.1
 - video_converter: 1.0.4
 - viewer: 1.6.0
 - workflow_media_converter: 1.3.3
 - workflow_ocr: 1.22.4
 - workflow_pdf_converter: 1.7.0
 - workflowengine: 2.4.0
Disabled:
 - circles: 22.1.1
 - encryption
 - files_trashbin: 1.11.0
 - files_versions: 1.14.0
 - firstrunwizard: 2.11.0
 - music: 1.5.1
 - recommendations: 1.1.0
 - survey_client: 1.10.0
 - user_ldap
 - user_status: 1.1.1
 - weather_status: 1.1.0

</details>

<details><summary>Configuration (config/config.php)</summary>

{
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "secret": "***REMOVED SENSITIVE VALUE***",
    "trusted_domains": [
        "localhost",
        "a.loc.al.ip",
        "cloud.mydomain.tld",
        "cloud.mydomain2.tld"
    ],
    "datadirectory": "***REMOVED SENSITIVE VALUE***",
    "dbtype": "pgsql",
    "version": "22.2.4.3",
    "overwrite.cli.url": "https:\/\/cloud.mydomain.tld\/",
    "dbname": "***REMOVED SENSITIVE VALUE***",
    "dbhost": "***REMOVED SENSITIVE VALUE***",
    "dbport": "",
    "dbtableprefix": "oc_",
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "installed": true,
    "instanceid": "***REMOVED SENSITIVE VALUE***",
    "upgrade.disable-web": true,
    "log_type": "file",
    "logfile": "\/var\/log\/nextcloud\/nextcloud.log",
    "loglevel": "2",
    "log.condition": {
        "apps": [
            "admin_audit"
        ]
    },
    "mail_smtpmode": "smtp",
    "remember_login_cookie_lifetime": "1800",
    "log_rotate_size": "10485760",
    "trashbin_retention_obligation": "auto, 60",
    "versions_retention_obligation": "auto, 180",
    "activity_expire_days": "120",
    "simpleSignUpLink.shown": false,
    "memcache.local": "\\OC\\Memcache\\Redis",
    "filelocking.enabled": true,
    "memcache.distributed": "\\OC\\Memcache\\Redis",
    "memcache.locking": "\\OC\\Memcache\\Redis",
    "redis": {
        "host": "***REMOVED SENSITIVE VALUE***",
        "port": 0,
        "timeout": 0.5,
        "dbindex": 0,
        "password": "***REMOVED SENSITIVE VALUE***"
    },
    "default_phone_region": "de",
    "logtimezone": "Europe\/Berlin",
    "default_language": "de",
    "default_locale": "de_DE",
    "allow_user_to_change_display_name": true,
    "htaccess.RewriteBase": "\/",
    "maintenance": false,
    "mail_sendmailmode": "smtp",
    "mail_from_address": "***REMOVED SENSITIVE VALUE***",
    "mail_domain": "***REMOVED SENSITIVE VALUE***",
    "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpport": "587",
    "mail_smtpsecure": "tls",
    "app.mail.verify-tls-peer": false,
    "mail_smtpauthtype": "LOGIN",
    "mail_smtpauth": 1,
    "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
    "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
    "has_rebuilt_cache": true,
    "trusted_proxies": "***REMOVED SENSITIVE VALUE***"
}

</details>

External storages: yes

<details><summary>External storage configuration</summary>

+----------+-----------------+----------+-----------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------------------------------+------------------+-------------------+----------+
| Mount ID | Mount Point     | Storage  | Authentication Type   | Configuration                                                                                                                                                 | Options                               | Applicable Users | Applicable Groups | Type     |
+----------+-----------------+----------+-----------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------------------------------+------------------+-------------------+----------+
| 4        | /Fotos          | SMB/CIFS | Username and password | share: "osna\/Fotos", root: "", domain: "", show_hidden: false, check_acl: false, timeout: "", user: "dingdong", password: "***", host: "192.168.192.199"        | readonly: false, enable_sharing: true | dongdong          |                   | Personal |
| 2        | /extHDD/O-Drive | SMB/CIFS | Username and password | host: "internal ip", share: "osna\/users\/office", root: "", domain: "", show_hidden: true, check_acl: false, timeout: "", user: "dingdong", password: "***" | enable_sharing: true, readonly: false | dongdong          |                   | Personal |
| 5        | /Fotos          | SMB/CIFS | Username and password | host: "internal IP", share: "osna\/Fotos", root: "", domain: "", show_hidden: false, check_acl: false, timeout: "", user: "dingdong", password: "***"        | enable_sharing: true, readonly: false | dingding         |                   | Personal |
| 6        | /Fotos          | SMB/CIFS | Username and password | host: "internal IP", share: "osna\/Fotos", root: "", domain: "", show_hidden: false, check_acl: false, timeout: "", user: "dingdong", password: "***"        | enable_sharing: true, readonly: false | dingding         |                   | Personal |
| 8        | /extHDD/N-Drive | Local    | None                  | datadir: "\/mnt\/smbshares\/1"                                                                                                                                | readonly: true, enable_sharing: true  |                  | admin, Benutzer   | Admin    |
+----------+-----------------+----------+-----------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------------------------------+------------------+-------------------+----------+

</details>

Encryption: no

User-backends:

OC\User\Database

OCA\User_SAML\UserBackend

Talk configuration:

STUN servers

cloud.mydomain.tld:3478

TURN servers

turn:cloud.mydomain.tld:3478 - udp,tcp

Signaling servers (mode: internal):

no custom server configured

Browser: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:97.0) Gecko/20100101 Firefox/97.0

```
Login as admin user into your Nextcloud and access 
http://example.com/index.php/settings/integrity/failed 
"No errors have been found"
```
### Nextcloud log
<html><body>

Error | PHP | Allowed  memory size of 536870912 bytes exhausted (tried to allocate 406881344  bytes) at /var/www/nextcloud/apps/dav/lib/CardDAV/CardDavBackend.php#972
-- | -- | --

</body>
</html>
```
Insert your Nextcloud log here
```

```


if there's some info missing pls don't hesitate and let me know so that I could add itr

**Comment by as01p** [comment]
Same problem, everything is good in the database and a synchronization (add/remove) seems to be correct (from desktop to laptop). But nothing is visible in the web interface anymore.

**Comment by ChristophWurst** [comment]
if the trace is right then the process runs out of memory at https://github.com/nextcloud/server/blob/eb0d4efdc34ec45ce7d2df4627c193e88be72040/apps/dav/lib/CardDAV/CardDavBackend.php#L972. That is when a card is read from the DB. Do you have any records with exceptionally large data? 

```
MariaDB [nextclouddev]> select id, LENGTH(carddata) FROM oc_cards ORDER BY LENGTH(carddata) DESC LIMIT 10;
+----+------------------+
| id | LENGTH(carddata) |
+----+------------------+
| 48 |            31350 |
|  3 |            26810 |
|  1 |            23353 |
|  2 |            22592 |
|  4 |             9538 |
|  9 |             9538 |
| 17 |             9538 |
| 18 |             9538 |
| 20 |             9538 |
| 21 |             9538 |
+----+------------------+
10 rows in set (0.010 sec)
```

^ on my dev instance. Largest card is 31KB.

**Comment by JimmyKater** [comment]
@ChristophWurst 
As I run PostgreSQL my output looks a bit different. And yes, apparently there's a very big item
id,length
457,406873121
593,8167355
276,7439258
546,3227431
886,1904221
952,1672325
480,1541315
532,1532940
189,1516768
1025,1504267

**Comment by ChristophWurst** [comment]
this is probably some sensitive data. can I ask you to look into the row with id 457 and check if it's a valid card? maybe you can see what uses so much data in there. my educated guess would be a huge contact photo. Yet I can't imagine anyone uploaded 388MB for a contact photo …

**Comment by JimmyKater** [comment]
ummm I would share the data with you but I can't access id457 correctly. Most prolly I'm running into a timeout while trying to access the data there (note, I'm using Adminer to access the DB). 

I really can't remeber having loaded up ANY contactpicture to one of my contacts... but wells maybe it was made by some automatic routine from NC (but that would be a different story then). 

So how to proceed from here?

**Comment by ChristophWurst** [comment]
you could use the postgres `substring` function to only get the first 1000bytes.that might already give you an idea of the data.

**Comment by JimmyKater** [comment]
@ChristophWurst `substring` seems to be working only with `char` while the only data I have from that id is the id number 457...

**Comment by ChristophWurst** [comment]
``SELECT `substring(carddata, 1000) FROM oc_cards WHERE id=457`` doesn't work?

**Comment by JimmyKater** [comment]
@ChristophWurst thanks for that command. I'm not into data-bases... :-( 

though I could enter it (adding a blank between `substring` and `(`  ;-) ) it again ran into a timeout and I received no answer. Even not after shrinking it down to 500. 

I think I'd like to delete it and then see what happens. As I have a copy of the orginal version of all adresses it's no bniggie. I have searched there for any big entries... and there are none. So it might have been caused by something else. Though all in all there are roughly 1050 adresses in my database.

How to proceed? Since I think this is a surgery on an open chest I'd like to be really sure about what to do next, exactly.

**Comment by ChristophWurst** [comment]
It *should* be save to delete this one row from the database. I'm not an expert with the carddav code, though.

**Comment by ChristophWurst** [comment]
> Same problem

@as01p could you please check if you do indeed have the *same* problem and a row with huge data?

**Comment by as01p** [comment]
7388	65366320
7015	1775808
7178	1775808
7115	807300
7277	807300
7355	703534
7331	688440
7103	619088
7265	619088
7289	570244

**Comment by JimmyKater** [comment]
@as01p 

ummm... I am unsure for the right syntax of how to delete an item correctly from my DB. Would you be so kind and eventually tell me how to do that safely? That would be awesome!

**Comment by JimmyKater** [comment]
is it `DELETE FROM oc_cards WHERE id=457 count; `?

**Comment by as01p** [comment]
DELETE FROM oc_cards WHERE id=457;
commit;

**Comment by JimmyKater** [comment]
@ChristophWurst with a little help from @as01p I managed to deleted said item. And now it seems to be working again. So this problem is solved.

Thanks for your help and time... all of you guys! <3<3<3

**Comment by remytms** [comment]
Deleting the biggest card from the table `oc_cards` indeed fixes the issue. But, looking at the size of other cards, I realized that there is other card that where very big.

The data in the `carddata` seams to be raw data that cannot been read by a human. So in order to look at the biggest cards and fix them, I used the `uid` of theses card.

I put the uid in the url and I can there fix the card by removing the image of the contact. Mine looks like this: https://foo.bar.org/apps/contacts/Tous les contacts/**a97a82d9-3c34-4af4-9725-d1e4298dff04**~contacts

The image was an automatic download by either my phone, or nextcloud. But this picture was way to large (about 20Mo).
